app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,13 @@
+module Foreman
+  class Deprecation
+    #deadline_version - is the version the deprecation is going to be deleted, the format must be a major release e.g "1.8"
+    def self.deprecation_warning(foreman_version_deadline, info, caller)
```

Comment:
Rather than passing `caller` through, just try passing `caller(2)` into AS::Deprecation.warn, which should chop off the first line of the stack so it accurately shows the original caller.